lib/rucio/core/replica.py

```diff
@@ -2678,43 +2678,49 @@ def list_dataset_replicas(scope, name, deep=False, session=None):
+                                  models.CollectionReplica.name.in_(names_by_intscope[scope])))
 
-    try:
-        # chunk size refers to the number of different scopes, see above
-        for chunk in chunks(condition, 10):
-            query = session.query(models.CollectionReplica.scope,
-                                  models.CollectionReplica.name,
-                                  models.RSE.rse,
-                                  models.CollectionReplica.rse_id,
-                                  models.CollectionReplica.bytes,
-                                  models.CollectionReplica.length,
-                                  models.CollectionReplica.available_bytes,
-                                  models.CollectionReplica.available_replicas_cnt.label("available_length"),
-                                  models.CollectionReplica.state,
-                                  models.CollectionReplica.created_at,
-                                  models.CollectionReplica.updated_at,
-                                  models.CollectionReplica.accessed_at) \
-                .filter(models.CollectionReplica.did_type == DIDType.DATASET) \
-                .filter(models.CollectionReplica.rse_id == models.RSE.id) \
-                .filter(or_(*chunk)) \
-                .filter(models.RSE.deleted == false())
-            for row in query:
-                yield row._asdict()
-    except NoResultFound:
-        raise exception.DataIdentifierNotFound('No Data Identifiers found')
+        try:
+            # chunk size refers to the number of different scopes, see above
+            for chunk in chunks(condition, 10):
+                query = session.query(models.CollectionReplica.scope,
+                                      models.CollectionReplica.name,
+                                      models.RSE.rse,
+                                      models.CollectionReplica.rse_id,
+                                      models.CollectionReplica.bytes,
+                                      models.CollectionReplica.length,
+                                      models.CollectionReplica.available_bytes,
+                                      models.CollectionReplica.available_replicas_cnt.label("available_length"),
+                                      models.CollectionReplica.state,
+                                      models.CollectionReplica.created_at,
+                                      models.CollectionReplica.updated_at,
+                                      models.CollectionReplica.accessed_at) \
+                    .filter(models.CollectionReplica.did_type == DIDType.DATASET) \
+                    .filter(models.CollectionReplica.rse_id == models.RSE.id) \
+                    .filter(or_(*chunk)) \
+                    .filter(models.RSE.deleted == false())
+                for row in query:
+                    yield row._asdict()
+        except NoResultFound:
```

Comment:
I am not sure if this part would ever be raised? Did you test this? If yes, there is definitely a difference then if you run it with `deep=True` - not sure if this would be raised in that case. So the method acts a bit different for empty results; Would be worth to check.

Reply:
I do not think these are my changes. I only changed indentation for this fragment

Reply:
Yes I understand, sorry should have phrased this more clearly. I just think that due to this part there is a difference if there are no results between `deep=True` and `deep=False`, which is a bit inconsistent. So I am wondering if it is not better to remove this except completely. (Not sure if it is raised at all in the first place.

Reply:
Do you mean that if there are no results, then it should simply yield nothing in both cases (deep=True and deep=False), instead of generating the NotFound exception ?

I would agree to that actually. I think "no results" is a valid result in this case. But I wonder if maybe at some place, someone expects this exception ?

Reply:
Exactly it should just yield nothing.
I suspect that `NoResultFound` is not even raised in this case actually, but I think we can relatively safely remove that exception.